### PR TITLE
K8SPXC-682: Adjust innodb_buffer_pool_chunk_size to respect memory limit

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPXC-682](https://jira.percona.com/browse/K8SPXC-682) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | failed |
| scaling | failed |
| recreate | failed |
| upgrade-haproxy | failed |
| big-data | failed |
| storage | failed |
| scheduled-backup | failed |
| init-deploy | failed |
| scaling-proxysql | failed |
| self-healing | failed |
| restore-to-encrypted-cluster | failed |
| upgrade-proxysql | failed |
| limits | failed |
| security-context | failed |
| demand-backup | failed |
| self-healing-advanced | failed |
| smart-update | failed |
| monitoring-2-0 | failed |
| pitr | failed |
| operator-self-healing | failed |
| upgrade-consistency | failed |
| affinity | failed |
| demand-backup-encrypted-with-tls | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |
| validation-hook | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/882/commits/c1bc9ee923a7205de53e4b6483a0b8e65c48199e
image: `perconalab/percona-xtradb-cluster-operator:PR-882-c1bc9ee9`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| recreate | passed |
| scaling | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/882/commits/c1bc9ee923a7205de53e4b6483a0b8e65c48199e
image: `perconalab/percona-xtradb-cluster-operator:PR-882-c1bc9ee9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| demand-backup | passed |
| limits | passed |
| pitr | passed |
| monitoring-2-0 | passed |
| demand-backup-encrypted-with-tls | passed |
| affinity | passed |
| scaling | passed |
| one-pod | passed |
| scaling-proxysql | passed |
| auto-tuning | passed |
| security-context | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| scheduled-backup | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/882/commits/c1bc9ee923a7205de53e4b6483a0b8e65c48199e
image: `perconalab/percona-xtradb-cluster-operator:PR-882-c1bc9ee9`

File: pkg/pxc/app/config/autotune.go [hunk 1]
```diff
@@ -19,6 +19,18 @@ func getAutoTuneParams(memory string) (string, error) {
 	paramValue := "\n" + "innodb_buffer_pool_size" + " = " + poolSizeVal
 	autotuneParams += paramValue
 
+	// Adjust innodb_buffer_pool_chunk_size
+	// If innodb_buffer_pool_size is bigger than 1G, innodb_buffer_pool_instances is set to 8.
+	// By default, innodb_buffer_pool_chunk_size is 128M and innodb_buffer_pool_size needs to be
+	// multiple of innodb_buffer_pool_chunk_size * innodb_buffer_pool_instances.
+	// More info: https://dev.mysql.com/doc/refman/8.0/en/innodb-buffer-pool-resize.html
+	if poolSize > int64(1000000000) {
+		chunkSize := q.Value() / int64(8)
```
Comment (tplavcic) [inline pkg/pxc/app/config/autotune.go:28]: Does this mean that we cannot change `innodb_buffer_pool_instances` via config to anything other than 8?

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: pkg/pxc/app/config/autotune.go (hunk 1 above)
Comment (delgod) [reply pkg/pxc/app/config/autotune.go:28]: user can change value via `spec.pxc.configuration`

**Review by tplavcic — APPROVED** [review]

**Pull request merged**